### ISSUE TITLE
Bridge to kubernetes stopped working after VS2022 upgrade to 17.3.0

### ISSUE THREAD

**Issue by assi010** [Azure/Bridge-To-Kubernetes#22]
Describe the bug
Since upgrading Visual Studio 2022 to version 17.3.0 I am unable to debug using the bridge to kubernetes extension. 
While the debugging session will keep running in visual studio itself, it seems that everything on the cluster itself shuts down so none of the requests are routed to the local session anymore.

To Reproduce

Upgrade Visual Studio 2022 to version 17.3.0
Try to debug any service on a kubernetes cluster

Logs
[bridge-library-2022-08-19-14-42-02-76.txt](https://github.com/Azure/Bridge-To-Kubernetes/files/9383152/bridge-library-2022-08-19-14-42-02-76.txt)
[bridge-mindarocli-2022-08-19-14-42-01-76.txt](https://github.com/Azure/Bridge-To-Kubernetes/files/9383153/bridge-mindarocli-2022-08-19-14-42-01-76.txt)
[bridge-endpointmanager-2022-08-19-14-42-12-12920.txt](https://github.com/Azure/Bridge-To-Kubernetes/files/9383154/bridge-endpointmanager-2022-08-19-14-42-12-12920.txt)
[bridge-endpointmanagerlauncher-2022-08-19-14-42-12-1184.txt](https://github.com/Azure/Bridge-To-Kubernetes/files/9383155/bridge-endpointmanagerlauncher-2022-08-19-14-42-12-1184.txt)

Environment Details
Visual Studio: 17.3.0
Bridge to Kubernetes extension: 3.0.20220308.1
Operating System: Windows 10 Pro 21H1 19043.1826

In VS Code everything still works as expected.

**Comment by elenavillamil** [comment]
Hi, I see you are using march version of bridge, could you try with latest and see if it helps? I have tried running this on Visual Studio on Windows and it worked for me (screenshot below). I just follow steps [here](https://docs.microsoft.com/en-us/visualstudio/bridge/bridge-to-kubernetes-vs). I am using windows 11 Enterprise, VisualStudio 17.3.1 and b2b 3.0.20220308.1
![image](https://user-images.githubusercontent.com/4663400/186218635-7bb2f296-9edd-43e8-98c7-5733b28ed444.png)

**Comment by elenavillamil** [comment]
Are you using isolation? If so could you also provide routing manager logs?

**Comment by nfokkema** [comment]
Same issue since updating. It works fine on 17.2.*
Running a .net 6 webapi
Not running in isolation

**Bridge config**
![image](https://user-images.githubusercontent.com/105846043/186363570-97899323-0909-44ca-bb7c-ec89840f4d4d.png)

**Environment**
```text
Windows 10 Pro 21H2 19044.1466
Visual Studio 2022 17.3.2
Bridge to Kubernetes 2022 3.0.20220308.1
All freshly installed on a VM
```

**Logs**
[bridge-endpointmanager-2022-08-24-07-38-52-1904.txt](https://github.com/Azure/Bridge-To-Kubernetes/files/9414029/bridge-endpointmanager-2022-08-24-07-38-52-1904.txt)
[bridge-endpointmanagerlauncher-2022-08-24-07-38-51-1448.txt](https://github.com/Azure/Bridge-To-Kubernetes/files/9414030/bridge-endpointmanagerlauncher-2022-08-24-07-38-51-1448.txt)
[bridge-library-2022-08-24-07-38-43-536.txt](https://github.com/Azure/Bridge-To-Kubernetes/files/9414031/bridge-library-2022-08-24-07-38-43-536.txt)
[bridge-library-2022-08-24-07-38-47-4108.txt](https://github.com/Azure/Bridge-To-Kubernetes/files/9414032/bridge-library-2022-08-24-07-38-47-4108.txt)
[bridge-mindarocli-2022-08-24-07-38-42-536.txt](https://github.com/Azure/Bridge-To-Kubernetes/files/9414033/bridge-mindarocli-2022-08-24-07-38-42-536.txt)
[bridge-mindarocli-2022-08-24-07-38-47-4108.txt](https://github.com/Azure/Bridge-To-Kubernetes/files/9414034/bridge-mindarocli-2022-08-24-07-38-47-4108.txt)

The error that pops up when staring a debug session
![Schermafbeelding 2022-08-24 093930](https://user-images.githubusercontent.com/105846043/186362337-88a3af90-3d33-4ee6-9223-2a6d87f694e9.png)

```json
{
  "$schema": "http://json.schemastore.org/launchsettings.json",
  "profiles": {
    "DnaService.Api": {
      "commandName": "Project",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      },
      "dotnetRunMessages": "true",
      "applicationUrl": "http://localhost:5101"
    },
    "Bridge to Kubernetes": {
      "commandName": "AzureDevSpacesLocal"
    }
  }
}
```

This suggests that it somehow now ignores the defaults in the `launchsettings.json`. 
Setting `"launchBrowser": false` explicitly does nothing, and setting `"launchUrl": "http://localhost:5101"` explicitly also does nothing.

After the error dialog the debugging session is still running but is not connected to the cluster (no restore pod nor connectivity with cluster services)

Let know if I can provide further info or test something on my reproduction VM!

**Comment by assi010** [comment]
Thanks for the reply @elenavillamil. I experience the same issue with or without routing isolation and I can even hit the breakpoints when setting a breakpoint on the probing endpoints. 

The issue is that while locally everything _seems_ to run smoothly without any errors, on AKS suddenly all the pods related to bridge are suddenly removed without warning after a few seconds. So new requests are no longer routed to my local session.

This seems to be a different issue as what nfokkema is experiencing. @nfokkema I can reproduce your issue by removing the https entry from the applicationUrl in the `launchsettings.json`, so I would suggest trying it again with an https url set.

```json
{
  "$schema": "http://json.schemastore.org/launchsettings.json",
  "profiles": {
    "DnaService.Api": {
      "commandName": "Project",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      },
      "dotnetRunMessages": "true",
      "applicationUrl": "https://localhost:5100;http://localhost:5101"
    },
    "Bridge to Kubernetes": {
      "commandName": "AzureDevSpacesLocal"
    }
  }
}
```

**Comment by GHMeyer0** [comment]
I'm having the same problem, after starting the debug session it restores the services in kubernetes but keeps the debugger.

Below the container logs

`
2022-08-24T22:23:16.7550589Z | RestorationJob | TRACE | Waiting to restore previous state on Deployment /tiino-integrado-erp...\nOperation context: <json>{"clientRequestId":"badfac88-2726-4393-82e3-4f4253e2c7cc","correlationId":"ac5e30ca-cbd8-4050-b1d2-d06c3666d9e5:00357ba98147:13b3fc8eb174","requestId":null,"userSubscriptionId":null,"startTime":"2022-08-24T22:23:14.7704111Z","userAgent":"RestorationJob/1.0.20220817.1","requestHttpMethod":null,"requestUri":null,"version":"1.0.20220817.1","requestHeaders":{},"loggingProperties":{"applicationName":"RestorationJob","deviceOperatingSystem":"Linux 5.4.0-124-generic #140-Ubuntu SMP Thu Aug 4 02:23:37 UTC 2022","framework":".NET Core 3.1.16","processId":1,"targetEnvironment":"Production"}}</json>
2022-08-24T22:23:26.6638949Z | RestorationJob | TRACE | AssemblyLoadContext unloading
2022-08-24T22:23:29.0591091Z | RestorationJob | ERROR | Logging handled exception: System.ArgumentNullException: {"ClassName":"System.ArgumentNullException","Message":"Value cannot be null.","Data":null,"InnerException":null,"HelpURL":null,"StackTraceString":"   at k8s.AbstractKubernetes.ReadNamespacedDeploymentWithHttpMessagesAsync(String name, String namespaceParameter, Nullable`1 pretty, IReadOnlyDictionary`2 customHeaders, CancellationToken cancellationToken)\n   at k8s.KubernetesExtensions.ReadNamespacedDeploymentAsync(IBasicKubernetes operations, String name, String namespaceParameter, Nullable`1 pretty, CancellationToken cancellationToken)\n   at Microsoft.BridgeToKubernetes.Common.Kubernetes.KubernetesClient.<>c__DisplayClass63_0.<<GetV1DeploymentAsync>b__0>d.MoveNext() in /src/common/Kubernetes/KubernetesClient.cs:line 683\n--- End of stack trace from previous location where exception was thrown ---\n   at Microsoft.BridgeToKubernetes.Common.Kubernetes.KubernetesClient.<>c__DisplayClass75_0`1.<<ClientInvokeWrapperAsync>b__0>d.MoveNext() in /src/common/Kubernetes/KubernetesClient.cs:line 1058\n--- End of stack trace from previous location where exception was thrown ---\n   at Microsoft.BridgeToKubernetes.Common.Utilities.WebUtilities.RetryUntilTimeAsync(Func`2 action, TimeSpan maxWaitTime, CancellationToken cancellationToken) in /src/common/Utilities/WebUtilities.cs:line 112\n   at Microsoft.BridgeToKubernetes.Common.Kubernetes.KubernetesClient.ClientInvokeWrapperAsync[T](Func`1 handler, String operation, CancellationToken cancellationToken) in /src/common/Kubernetes/KubernetesClient.cs:line 1054\n   at Microsoft.BridgeToKubernetes.Common.Kubernetes.KubernetesClient.<>c__DisplayClass24_0.<<ListPodsForDeploymentAsync>b__0>d.MoveNext() in /src/common/Kubernetes/KubernetesClient.cs:line 202\n--- End of stack trace from previous location where exception was thrown ---\n   at Microsoft.BridgeToKubernetes.Common.Kubernetes.KubernetesClient.<>c__DisplayClass75_0`1.<<ClientInvokeWrapperAsync>b__0>d.MoveNext() in /src/common/Kubernetes/KubernetesClient.cs:line 1058\n--- End of stack trace from previous location where exception was thrown ---\n   at Microsoft.BridgeToKubernetes.Common.Utilities.WebUtilities.RetryUntilTimeAsync(Func`2 action, TimeSpan maxWaitTime, CancellationToken cancellationToken) in /src/common/Utilities/WebUtilities.cs:line 112\n   at Microsoft.BridgeToKubernetes.Common.Kubernetes.KubernetesClient.ClientInvokeWrapperAsync[T](Func`1 handler, String operation, CancellationToken cancellationToken) in /src/common/Kubernetes/KubernetesClient.cs:line 1054\n   at Microsoft.BridgeToKubernetes.DevHostAgent.RestorationJob.RestorationJobApp._GetAgentEndpointAsync(DeploymentPatch deploymentPatch, CancellationToken cancellationToken) in /src/devhostagent.restorationjob/RestorationJobApp.cs:line 236","RemoteStackTraceString":null,"RemoteStackIndex":0,"ExceptionMethod":null,"HResult":-2147467261,"Source":"KubernetesClient.Basic","WatsonBuckets":null,"ParamName":"namespaceParameter"}
2022-08-24T22:23:29.2606899Z | RestorationJob | TRACE | Event: RestorationJob-AgentPing <json>
`

`{
  "ClassName": "System.ArgumentNullException",
  "Message": "Value cannot be null.",
  "Data": null,
  "InnerException": null,
  "HelpURL": null,
  "StackTraceString": "   at k8s.AbstractKubernetes.ReadNamespacedDeploymentWithHttpMessagesAsync(String name, String namespaceParameter, Nullable`1 pretty, IReadOnlyDictionary`2 customHeaders, CancellationToken cancellationToken)\n   at k8s.KubernetesExtensions.ReadNamespacedDeploymentAsync(IBasicKubernetes operations, String name, String namespaceParameter, Nullable`1 pretty, CancellationToken cancellationToken)\n   at Microsoft.BridgeToKubernetes.Common.Kubernetes.KubernetesClient.<>c__DisplayClass63_0.<<GetV1DeploymentAsync>b__0>d.MoveNext() in /src/common/Kubernetes/KubernetesClient.cs:line 683\n--- End of stack trace from previous location where exception was thrown ---\n   at Microsoft.BridgeToKubernetes.Common.Kubernetes.KubernetesClient.<>c__DisplayClass75_0`1.<<ClientInvokeWrapperAsync>b__0>d.MoveNext() in /src/common/Kubernetes/KubernetesClient.cs:line 1058\n--- End of stack trace from previous location where exception was thrown ---\n   at Microsoft.BridgeToKubernetes.Common.Utilities.WebUtilities.RetryUntilTimeAsync(Func`2 action, TimeSpan maxWaitTime, CancellationToken cancellationToken) in /src/common/Utilities/WebUtilities.cs:line 112\n   at Microsoft.BridgeToKubernetes.Common.Kubernetes.KubernetesClient.ClientInvokeWrapperAsync[T](Func`1 handler, String operation, CancellationToken cancellationToken) in /src/common/Kubernetes/KubernetesClient.cs:line 1054\n   at Microsoft.BridgeToKubernetes.Common.Kubernetes.KubernetesClient.<>c__DisplayClass24_0.<<ListPodsForDeploymentAsync>b__0>d.MoveNext() in /src/common/Kubernetes/KubernetesClient.cs:line 202\n--- End of stack trace from previous location where exception was thrown ---\n   at Microsoft.BridgeToKubernetes.Common.Kubernetes.KubernetesClient.<>c__DisplayClass75_0`1.<<ClientInvokeWrapperAsync>b__0>d.MoveNext() in /src/common/Kubernetes/KubernetesClient.cs:line 1058\n--- End of stack trace from previous location where exception was thrown ---\n   at Microsoft.BridgeToKubernetes.Common.Utilities.WebUtilities.RetryUntilTimeAsync(Func`2 action, TimeSpan maxWaitTime, CancellationToken cancellationToken) in /src/common/Utilities/WebUtilities.cs:line 112\n   at Microsoft.BridgeToKubernetes.Common.Kubernetes.KubernetesClient.ClientInvokeWrapperAsync[T](Func`1 handler, String operation, CancellationToken cancellationToken) in /src/common/Kubernetes/KubernetesClient.cs:line 1054\n   at Microsoft.BridgeToKubernetes.DevHostAgent.RestorationJob.RestorationJobApp._GetAgentEndpointAsync(DeploymentPatch deploymentPatch, CancellationToken cancellationToken) in /src/devhostagent.restorationjob/RestorationJobApp.cs:line 236",
  "RemoteStackTraceString": null,
  "RemoteStackIndex": 0,
  "ExceptionMethod": null,
  "HResult": -2147467261,
  "Source": "KubernetesClient.Basic",
  "WatsonBuckets": null,
  "ParamName": "namespaceParameter"
}`

**Comment by qpetraroia** [comment]
Thanks folks, we will label this as a bug

**Comment by elenavillamil** [comment]
While we try to reproduce this on our end, could you switch to previous version of Visual Studio that was working for you?

**Comment by GHMeyer0** [comment]
I'll do the rollback and test in the previous version

**Comment by VaclavK** [comment]
I can confirm similar experience.. started exploring bridge and immediately noticed these issues

17.3 bridge pods killed quickly, VS studio happily thinking it is debugging
17.2 experience as expected both with and w/o routing isolation

**Comment by elenavillamil** [comment]
Hi everyone,

We have found the problem. It seems latest release of Visual Studio updates ServiceHub to prevent long running process. This is why with latest version we see bridge cleaning up pretty much as soon as it starts. We are working on a fix, but until it is ready, tested and released users will need to stay in VS 17.2. 

Thank you, and sorry for the inconvenience.

**Comment by elenavillamil** [comment]
Hi everyone,

PR was checked in and preliminary initial testing/validation passed. We are hoping to have an update to the marketplace extension by the end of this week.

Thank you,
Elena

**Comment by elenavillamil** [comment]
Hi everyone,

New extension has been released. Could you please confirm issues is fixed so I can go ahead and close this bug. 

Thank you and sorry for the inconvenience,
Elena

**Comment by VaclavK** [comment]
Hi,

just tested it with VS 17.3.4 and latest version of the K8 bridge extension

- w/o isolation I can now debug my service... I get a connection failure error message when attempting routing isolation

in 17.2 I get a different message now (don't remember seeing it before and recall being able to do it before)  but my service is indeed using dapr 

assuming that 17.3 and 17.2 error messages are meant to be the same? Will set a non dapr service to try routing isolation under 17.3

![image](https://user-images.githubusercontent.com/3881864/190362709-b5ee9fb4-2adb-46ef-8746-5dc7bb2e36cd.png)

**Comment by elenavillamil** [comment]
Hi @VaclavK,

Indeed dapr support is not there when isolation is enabled. It has never been there. That error message was updated to make errors more clear to the user. Update of VS extension in marketplace pulled the new CLI version as well which is why you see the new message now :)

Thank you,
Elena

**Comment by nfokkema** [comment]
I can confirm the extension works in 17.3.4 without the need of a browser. 
Just make sure you explicitly set the (new?) `launchBrowser` option in `Bridge to Kubernetes` to false:
```json
{
    "MyApplication": {
      "commandName": "Project",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      },
      "applicationUrl": "http://localhost:5000
    },
    "Bridge to Kubernetes": {
      "commandName": "AzureDevSpacesLocal",
      "launchBrowser": false
    }
}
```

Thank you @elenavillamil for the fix! 🥇

**Comment by qpetraroia** [comment]
Hi @nfokkema, 

If your issue is solved, we would love to close this ticket! Thanks!

**Comment by nfokkema** [comment]
@qpetraroia fine by me.
Maybe give it a day so the OP @assi010 can test it on their side as well?

**Comment by assi010** [comment]
I can also confirm the issues are solved on 17.3.4, thanks for the fix @elenavillamil 👍
